lnbits/wallets/corelightningrest.py

```diff
@@ -0,0 +1,241 @@
+            try:
+                data = r.json()
+                error_message = data["error"]
+            except Exception:
+                error_message = r.text
+
+            return InvoiceResponse(False, None, None, error_message)
+
+        data = r.json()
+        assert "payment_hash" in data
+        assert "bolt11" in data
+        # NOTE: use payment_hash when corelightning-rest updates and supports it
+        # return InvoiceResponse(True, data["payment_hash"], data["bolt11"], None)
+        return InvoiceResponse(True, label, data["bolt11"], None)
+
+    async def pay_invoice(self, bolt11: str, fee_limit_msat: int) -> PaymentResponse:
+        invoice = lnbits_bolt11.decode(bolt11)
+        fee_limit_percent = fee_limit_msat / invoice.amount_msat * 100
+        r = await self.client.post(
+            f"{self.url}/v1/pay",
+            data={
+                "invoice": bolt11,
+                "maxfeepercent": f"{fee_limit_percent:.11}",
+                "exemptfee": 0,  # so fee_limit_percent is applied even on payments
+                # with fee < 5000 millisatoshi (which is default value of exemptfee)
+            },
+            timeout=None,
+        )
+
+        if r.is_error or "error" in r.json():
+            try:
+                data = r.json()
+                error_message = data["error"]
+            except Exception:
+                error_message = r.text
+            return PaymentResponse(False, None, None, None, error_message)
+
+        data = r.json()
+
+        if data["status"] != "complete":
+            return PaymentResponse(False, None, None, None, "payment failed")
+
+        checking_id = data["payment_hash"]
+        preimage = data["payment_preimage"]
+        fee_msat = data["msatoshi_sent"] - data["msatoshi"]
+
+        return PaymentResponse(
+            self.statuses.get(data["status"]), checking_id, fee_msat, preimage, None
+        )
+
+    async def get_invoice_status(self, payment: Payment) -> PaymentStatus:
+        # get invoice bolt11 from checking_id
+        # corelightning-rest wants the "label" here....
+        # NOTE: We can get rid of all labels and use payment_hash when
+        # corelightning-rest updates and supports it
+        r = await self.client.get(
+            f"{self.url}/v1/invoice/listInvoices",
+            params={"label": payment.checking_id},
+        )
+        try:
+            r.raise_for_status()
+            data = r.json()
+
+            if r.is_error or "error" in data or data.get("invoices") is None:
+                raise Exception("error in cln response")
+            return PaymentStatus(self.statuses.get(data["invoices"][0]["status"]))
+        except Exception as e:
+            logger.error(f"Error getting invoice status: {e}")
+            return PaymentStatus(None)
+
+    async def get_payment_status(self, payment: Payment) -> PaymentStatus:
+        # corelightning-rest wants the "bolt11" here.... sigh
```

Comment:
what if, instead of changing up the whole wallets interface we simply get the bolt11 from the database using the checking id? This wouldnt work for payments we did not initate, but thats never done anyways. Not ideal, but I guess its fine until cln-rest supports it
```suggestion
    async def get_invoice_status(self, checking_id: str) -> PaymentStatus:
        payment = await get_standalone_payment(checking_id)
        if not payment:
            raise ValueError("Unknown checking_id")
```